### PR TITLE
Add unit tests for com.zscat.mall.portal.util.CharUtil

### PR DESCRIPTION
**Pull request by JohnLBergqvist**
I've analysed your codebase and noticed that `com.zscat.mall.portal.util.CharUtil` is not fully tested.
I've written some tests for the methods in this class with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes that you consider important.

### CODE REVIEW

**Review by freddytuxworth — APPROVED** [review]

File: mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    PowerMockito.when(random.nextInt(anyInt())).thenReturn(32);
+    PowerMockito.whenNew(Random.class).withNoArguments().thenReturn(random);
+
+    Assert.assertEquals("6", CharUtil.getRandomString(1));
+    Assert.assertEquals("", CharUtil.getRandomString(0));
+  }
+
+  @Test
+  public void testPadLeft() {
+    Assert.assertEquals("a\'b\'c\u0000", CharUtil.padLeft("a\'b\'c", 6, '\u0000'));
```
Comment (petersear) [inline mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java]: line > 80 chars

File: mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    Assert.assertEquals("1", CharUtil.padLeft("1", 0, '\u0000'));
+  }
+
+  @Test
+  public void testPadRight() {
+    Assert.assertEquals("\u0000a\'b\'c", CharUtil.padRight("a\'b\'c", 6, '\u0000'));
```
Comment (petersear) [inline mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java]: line > 80 chars

File: mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public class CharUtilTest {
+  @PrepareForTest({CharUtil.class})
```
Comment (petersear) [inline mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java:17]: Blank line before `@PrepareForTest`?

**Review by petersear — APPROVED** [review]

File: mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java (hunk 3 above)
Comment (JohnLBergqvist) [reply mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java:17]: @petersear Done

File: mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java (hunk 1 above)
Comment (JohnLBergqvist) [reply mall-portal/src/test/java/com/macro/mall/portal/util/CharUtilTest.java]: Done. Even if I think this 80 char restriction is pointless.

**Pull request closed without merging**